### PR TITLE
Alerts for Redis and Elasticsearch

### PR DESCRIPTION
**Pull request by dzaslavskiy**
Alert for Redis key size
alert for es shard count
alert for es disk space

### CODE REVIEW

File: prometheus/rules.yml [hunk 1]
```diff
@@ -158,3 +158,35 @@ groups:
           summary: Watchtower failed when validating CF App configurations
           description: Check the Watchtower logs for failure details.
           icon: ❓
+
+  - name: ElasticsearchAlerts
+    rules:
+      - alert: High disk utilization
+        expr: elasticsearch_filesystem_data_available_bytes / elasticsearch_filesystem_data_size_bytes < 0.2
```
Comment (folksgl) [inline prometheus/rules.yml:165]: I think this reads more clearly if we reverse it to be typical fraction `(used space / total space available) > 0.8`. Note - I have not ran/tested this query.
```suggestion
        expr: elasticsearch_filesystem_data_size_bytes / elasticsearch_filesystem_data_available_bytes > 0.8
```

**Review by folksgl — CHANGES_REQUESTED** [review]

File: prometheus/rules.yml (hunk 1 above)
Comment (dzaslavskiy) [reply prometheus/rules.yml:165]: that is not how math works. you inverted one size of the equation so you have to invert the other size:
        expr: elasticsearch_filesystem_data_size_bytes / elasticsearch_filesystem_data_available_bytes > 5
I am not sure I like that better.

Comment (folksgl) [reply prometheus/rules.yml:165]: Shouldn't `data_size / space_available` always return a fraction indicating the % of space used? I'm confused now.

Comment (dzaslavskiy) [reply prometheus/rules.yml:165]: As written the expression is `free space / total space`

Comment (folksgl) [reply prometheus/rules.yml:165]: Oh! I didn't realize that `elasticsearch_filesystem_data_available_bytes` was the amount of space **left**. I was thinking about it as the amount of space that has been used.

The elasticsearch exporter docs have the following:
```
elasticsearch_filesystem_data_available_bytes	gauge	1	Available space on block device in bytes
elasticsearch_filesystem_data_free_bytes	gauge	1	Free space on block device in bytes
elasticsearch_filesystem_data_size_bytes	gauge	1	Size of block device in bytes
```
The descriptions of these metrics leave much to be desired. How does `data_available_bytes` differ from `data_size_bytes`?

**Review by folksgl — APPROVED** [review]

File: prometheus/rules.yml (hunk 1 above)
Comment (folksgl) [reply prometheus/rules.yml:165]: Reviewed this, my assumptions were incorrect on the meaning of the metrics. The logic here is sound.

**Pull request merged**